### ISSUE TITLE
Dynamically install required package manager version at runtime

### ISSUE THREAD

**Issue by JamieMagee** [dependabot/dependabot-core#9286]
### Code improvement description

Currently, we define static versions of each programming language or package manager that Dependabot uses at runtime. For example, [`npm`][1], [`rust`][2], [`python`][3], etc.

Unfortunately, this means that Dependabot can sometimes be a bottleneck for users who upgrade to a newer version that we don't currently install in our container images. For example see #9249.

We could remove this bottleneck, and source of user frustration, by dynamically installing the required version of a programming language at runtime if a specific version is detected in the repository. For .NET, that would be looking in `global.json`; for `npm` that would be looking at the `engines` property in `package.json, etc.
 
[1]: https://github.com/dependabot/dependabot-core/blob/7e6d6f13cdbebacbc93d26c0a9653c76bb268bde/npm_and_yarn/Dockerfile#L18
[2]: https://github.com/dependabot/dependabot-core/blob/7e6d6f13cdbebacbc93d26c0a9653c76bb268bde/cargo/Dockerfile#L1
[3]: https://github.com/dependabot/dependabot-core/blob/7e6d6f13cdbebacbc93d26c0a9653c76bb268bde/python/Dockerfile#L3-L7

**Comment by yeikel** [comment]
What should happen if the property is  not available? Fallback?

For `npm` for example, `engines` is optional

**Comment by JamieMagee** [comment]
Yeah, I think this should be in addition to the current default version that is installed in the container images.

**Comment by yeikel** [comment]
This is still very relevant. Ie: https://github.com/dependabot/dependabot-core/issues/14139
